### PR TITLE
Distributed pkg

### LINKED ISSUE #50: Distributed package storage
Today the remote registry is a json file defined by config `command_repository_base_url`/index.json.

This index file contains the packages information:
```json
{
  name: "devenv",
  version: "1.0.48071",
  checksum: "xxx",
  startPartition: 0,
  endPartition: 9
}
```
The package itself is located at `command_repository_base_url`/<pkg_name>/<pkg_name>_<pkg_version>.pkg

Such setting requires all packages to release to the specific server.

By introducing a URL field into the package metadata, we can simplify the release process: calculating the checksum, and update the index.json file.

### LINKED ISSUE #53: Add --json option in config built-in command
Today the config command output is in plain text. The integration test uses grep to check if a field is present. If we support a json output format, it will allow us to make more advanced test.

### PR DESCRIPTION
**Pull request by bhou-crto**
close #50 
close #53

### CODE REVIEW

File: test/integration.sh
Comment (gmonceyron) [inline test/integration.sh:175]: Not sure this is necessary

File: examples/remote-config/remote_config.json
Comment (gmonceyron) [inline examples/remote-config/remote_config.json:2]: Once this is merged, you will need another review to change the branch you are targeting for tests before deleting this branch, right ?

File: test/packages-src/bonjour/manifest.mf
Comment (gmonceyron) [inline test/packages-src/bonjour/manifest.mf:1]: Is it worth keeping this in sources as it is also contained by the bonjour-v1.pkg file, and it is not used as a standalone file by tests ?

File: test/packages-src/bonjour/bonjour.bat
Comment (gmonceyron) [inline test/packages-src/bonjour/bonjour.bat:1]: Is it worth keeping this in sources as it is also contained by the bonjour-v1.pkg file, and it is not used as a standalone file by tests ?

File: test/integration.sh
Comment (bhou-crto) [reply test/integration.sh:7]: Yes, you are right, the branch contains files that required for integration test. We don't need to change it soon, as long as we don't delete the branch. But I will  prepare a PR once it is merged

File: examples/remote-config/remote_config.json
Comment (bhou-crto) [reply examples/remote-config/remote_config.json:2]: Yes

File: test/packages-src/bonjour/manifest.mf
Comment (bhou-crto) [reply test/packages-src/bonjour/manifest.mf:1]: I think so based on 2 reasons:
1. with the source code, it make us easier to make further changes in the pkg. 
2. we can use the source as a dropin in the test as well.

File: test/integration.sh
Comment (bhou-crto) [reply test/integration.sh]: Fixed

**Review by gmonceyron — APPROVED** [review]

**Pull request merged**
